pilot/pkg/xds/ads.go

```diff
@@ -554,6 +554,9 @@ func (s *DiscoveryServer) initProxyMetadata(node *core.Node) (*model.Proxy, erro
 	if err != nil {
 		return nil, err
 	}
+	if len(meta.Labels) == 0 {
```

Comment:
No labels is fine valid - it is different to have empty labels and unset labels

Reply:
>  it is different to have empty labels and unset labels

How do we distinguish? If metadata does not have labels - doesn't it still cause some functionality like sidecar not function correctly?

Reply:
Sidecar functions perfectly fine - you cannot select the pod as it has no labels. Just like Service, etc.

I am worried about someone getting spammed with logs when they have perfectly valid pods

Reply:
I was referring to Sidecar created with Workloadselectors - May be we should warn in sidecar code when it finds sidecar with workload selector in namespace but proxy does not have labels.

I will remove it from here

Reply:
That is also valid, you can have some pods with labels and some pods without labels in the same namespace?

Reply:
Probably. But since the pods in the namespace are hopefully fewer - it does not spam that much in the sidecar logic. 

Are the other changes are OK? If yes. I can fix the tests.